test/Feature/HLSLLib/distance.fp16.test

```diff
@@ -0,0 +1,84 @@
+#--- source.hlsl
+
+StructuredBuffer<half4> X : register(t0);
+StructuredBuffer<half4> Y : register(t1);
+
+RWStructuredBuffer<half> Result : register(u2);
+
+[numthreads(1,1,1)]
+void main() {
+  // distance ({0.0}, {3.0}) = 3.0
+  half R0 = distance(X[0].x, Y[0].x);
+  Result[0] = R0;
+  // distance({0.0, 0.0}, {3.0, 4.0}) = 5.0
+  half R1 = distance(X[0].xy, Y[0].xy);
+  Result[1] = R1;
+  // distance({0.0, 0.0, 0.0}, {3.0, 4.0, 12.0}) = 13.0
+  half R2 = distance(X[0].xyz, Y[0].xyz);
+  Result[2] = R2;
+  // distance({0.0, 0.0, 0.0, 0.0}, {3.0, 4.0, 12.0, 84.0}) = 85.0
+  half R3 = distance(X[0].xyzw, Y[0].xyzw);
+  Result[3] = R3;  
+}
+
+//--- pipeline.yaml
+
+---
+Shaders:
+  - Stage: Compute
+    Entry: main
+    DispatchSize: [1, 1, 1]
+Buffers:
+  - Name: X
+    Format: Float16
+    Stride: 8
+    Data: [ 0x0000, 0x0000, 0x0000, 0x0000 ]
+  - Name: Y
+    Format: Float16
+    Stride: 8
+    Data: [ 0x4200, 0x4400, 0x4a00, 0x5540 ]
+  - Name: Result
+    Format: Float16
+    Stride: 2
+    ZeroInitSize: 8
+  - Name: ExpectedResult
+    Format: Float16
+    Stride: 2
+    Data: [ 0x4200, 0x4500, 0x4a80, 0x5550 ]
+Results:
+  - Result: CheckResult
+    Rule: BufferFloatULP
+    ULPT: 1
+    Actual: Result
+    Expected: ExpectedResult
+DescriptorSets:
+  - Resources:
+    - Name: X
+      Kind: StructuredBuffer
+      DirectXBinding:
+        Register: 0
+        Space: 0
+      VulkanBinding:
+        Binding: 0
+    - Name: Y
+      Kind: StructuredBuffer
+      DirectXBinding:
+        Register: 1
+        Space: 0
+      VulkanBinding:
+        Binding: 1
+    - Name: Result
+      Kind: RWStructuredBuffer
+      DirectXBinding:
+        Register: 2
+        Space: 0
+      VulkanBinding:
+        Binding: 2
+...
+#--- end
+
+# UNSUPPORTED: Clang-Vulkan

```

Comment:
Why is Clang-Vulkan unsupported? Sorry if this has been explained before and I forgot or missed it.

Reply:
Can you leave a comment about that in the tests?